processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -71,37 +84,61 @@
 
   private String outOfRowsReason = null;
 
+  private final File tmpDirLocation;
+
   public OffheapIncrementalIndex(
       IncrementalIndexSchema incrementalIndexSchema,
       boolean deserializeComplexMetrics,
       boolean reportParseExceptions,
       boolean sortFacts,
-      int maxRowCount,
-      StupidPool<ByteBuffer> bufferPool
+      long maxOffheapSize
   )
   {
     super(incrementalIndexSchema, deserializeComplexMetrics, reportParseExceptions, sortFacts);
-    this.maxRowCount = maxRowCount;
-    this.bufferPool = bufferPool;
+    tmpDirLocation = Files.createTempDir();
+
+    try {
+      File mapDbFile = File.createTempFile("druid-offheap-index-mapdb", "", tmpDirLocation);
+      File aggsBBFile = File.createTempFile("druid-offheap-index-aggs", "", tmpDirLocation);
+
+      AtomicLong availableTotalSpace = new AtomicLong(maxOffheapSize);
+
+      final DBMaker dbMaker = new MyDBMaker(mapDbFile, 24, 29, availableTotalSpace)
```

Comment:
based on a bit of experimentation only also it is mapdb value in general, creating new buffers of 512M for storing data.
dint want to have too many confusing user visible config, this should be ok.